### PR TITLE
ZEPPELIN-3865. Paragraph cancel is broken

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?
Paragraph cancel is broken due to previous code refactoring, This PR fix it in straightforward way and add system test as well.

### What type of PR is it?
[Bug Fix ]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://jira.apache.org/jira/browse/ZEPPELIN-3865

### How should this be tested?
* CI pass

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Paragraph.java [hunk 1]
```diff
@@ -461,9 +461,10 @@ protected boolean jobAbort() {
     if (interpreter == null) {
       return true;
     }
-    Scheduler scheduler = interpreter.getScheduler();
-    if (scheduler == null) {
-      return true;
+    try {
+      interpreter.cancel(getInterpreterContext(null));
+    } catch (InterpreterException e) {
+      throw new RuntimeException(e);
```
Comment (felixcheung) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Paragraph.java]: why can't we throw InterpreterException here?

Comment (zjffdu) [reply zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Paragraph.java]: This is due to signature of class `Job`, change the signature of class `Job` will result in many changes in its callers (I will do this kind of refactoring later),  so I just throw RuntimeException here, but the exception would still be caught by ZeppelinSever.

File: zeppelin-server/src/test/java/org/apache/zeppelin/rest/ZeppelinSparkClusterTest.java [hunk 2]
```diff
@@ -176,6 +176,17 @@ public void scalaOutputTest() throws IOException {
     // test code completion
     List<InterpreterCompletion> completions = note.completion(p.getId(), "sc.", 2);
     assertTrue(completions.size() > 0);
+
+    // test cancel
+    p.setText("%spark sc.range(1,10).map(e=>{Thread.sleep(1000); e}).collect()");
+    note.run(p.getId(), false);
+    // wait for 3 seconds so that job is started before cancel
+    Thread.sleep(3000);
```
Comment (aka-demik) [inline zeppelin-server/src/test/java/org/apache/zeppelin/rest/ZeppelinSparkClusterTest.java]: Can you use busy waiting, event or latch here? In the current implementation, this test lasts for six seconds, even if it is not necessary.
Unfortunately, the tests are already slow = (

Comment (zjffdu) [reply zeppelin-server/src/test/java/org/apache/zeppelin/rest/ZeppelinSparkClusterTest.java]: Thanks @aka-demik , I have fixed it

**Pull request closed without merging**